cron/entrypoint.sh

```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/env bash
 
+if [ "$(ls -A /usr/local/share/ca-certificates/)" ]; then
+  update-ca-certificates
+fi
```

Comment:
Shouldn't this change be reflected in Sentry entrypoint too?

Also, do you think we can test this somehow? Maybe with a fake CA we use for tests or something? I don't want this feature to get broken in the future without us noticing.

Reply:
> Shouldn't this change be reflected in Sentry entrypoint too?
That's the point of the getsentry/sentry#27062 PR, and since the cron Dockerfile uses a dynamic base image, it overrides the entrypoint of the parent image and uses it's own script as the entrypoint, putting it in here would make sure any cron jobs get the same treatment for the CA roots in case they need to reach out.

> Also, do you think we can test this somehow? Maybe with a fake CA we use for tests or something?
Fair idea. The short of it would be to generate a CA root, do up a otherwise valid certificate signed by it, then spin up a HTTPS server and have Sentry try poking it, all while being reliable since we can't trust any hostnames nor IPs.

There is `host.docker.internal` and `host-gateway`, but they are not portable with the Win/Mac Docker versions having only one and Linux the other. Probably could solve IP/DNS by grabbing the the gateway IP from the docker network config and having an override to stuff a hostname in. That leaves figuring out a reliable way of testing Sentry can poke a URL. Could cheap out and just spawn a shell that uses python and the request lib in the `web` container and pokes the test HTTPS server and spits error codes.

Ironically this opens up a much larger can of worms than the original PR does. Probably worth making a sub folder to move the added test files into, since this will probably not all nicely fit in the existing `test.sh` file.

Reply:
>That's the point of the getsentry/sentry#27062 PR, and since the cron Dockerfile uses a dynamic base image, it overrides the entrypoint of the parent image and uses it's own script as the entrypoint, putting it in here would make sure any cron jobs get the same treatment for the CA roots in case they need to reach out.

Ehh, sorry I missed the PR over at Sentry. Makes sense, that said we have an entrypoint override here in onpremise for Sentry so I'm thinking maybe it is cleaner to keep everything in this repo?

>There is host.docker.internal and host-gateway, but they are not portable with the Win/Mac Docker versions having only one and Linux the other. Probably could solve IP/DNS by grabbing the the gateway IP from the docker network config and having an override to stuff a hostname in. 

We can easily add a test service into the compose mix to be able to easily ping it. 

>That leaves figuring out a reliable way of testing Sentry can poke a URL. Could cheap out and just spawn a shell that uses python and the request lib in the web container and pokes the test HTTPS server and spits error codes.

I think this is a fair way to test it until we find a better way?

>Ironically this opens up a much larger can of worms than the original PR does. Probably worth making a sub folder to move the added test files into, since this will probably not all nicely fit in the existing test.sh file.

Agreed, willing to chip in with tests once we solidify the implementation part.

Reply:
> Makes sense, that said we have an entrypoint override here in onpremise for Sentry so I'm thinking maybe it is cleaner to keep everything in this repo?
I had overlooked that entrypoint as onpremise does not generate a custom image for sentry, but I still needed to make a PR to set the env vars in the image to get it to use the system CA bundle and the overridden entrypoint still invokes the original one. It ultimately made a little more sense to me to include updating the system bundle in the base image.

> We can easily add a test service into the compose mix to be able to easily ping it.
Probably the sane way to go about it, though I would think it would be more ideal to not mess with the compose config since that is kinda what is being tested. Probably fine though if we never do anything to the normal services. *Probably.*

> Agreed, willing to chip in with tests once we solidify the implementation part.
I have a vague idea on how I would set that up, but I'll chew on that after a late lunch. Breaking up `./test.sh` into at a pre-test script to isolate the test setup and all the other tests to their own logical unit would be a good start, and it's own PR.